### DISCUSSION TITLE
Create floppy image

### DISCUSSION THREAD

**Discussion by egaralmeida** [acemielektron/fddEMU#28] [Q&A]
Hi, I can't seem to find a way to correctly create FDD images so they can load in FDDEmu. FDDEmu always replies "no disk" on the display

I've been trying this on a raspberry pi:

dd bs=512 count=2880 if=/dev/zero of=disk1.img
mkfs.msdos disk1.img
(also tried mkfs.fat -F 12 disk1.img)

I then mount the image (sudo mount -o loop disk1.img /media/floppy) copy the files (so far tried created a new text file using nano), then unmount and copy my disk1.img to the microSD card)

The microsd card is read correctly from the SD arduino example using the same pinout as FDDEmu (lists all files), so I assume whatever I'm doing is incorrect to be readable by FDDEmu.

Thanks in advance

**Comment by acemielektron** [comment]
Hi Egar,

I haven't looked at fddEMU recently. But I remember testing binaries on
GitHub https://GitHub.com/acemielektron/fddEMU/releases

Are you using released binaries or recompiled yourself. If you recompiled
fddEMU recently it is possible that a new version of compiler might have
broken supplied petitfs, otherwise it should at least listed the files on
SD card on screen and also on serial console.

If you used released binaries try the older one. If it still doesn't work I
might need to re test myself.

Looking forward to hearing from you,

acemielektron

**Reply by egaralmeida** [reply]
Thanks for your reply. I'm using the release for build a839e24, didn't compile myself.

**Comment by acemielektron** [comment]
Sorry for the late reply,
I had to re test release a839e24 binary (atmega328p-dual-flipped.hex) and
it loads floppy images from SD card without problem.
Petitfs lib used in fddEMU should support fat16 and fat32 formatted SD
cards.

acemielektron

**Comment by egaralmeida** [comment]
Thanks for checking that out, do you have a link for a floppy image known
to work?

On Wed, 22 Mar 2023 at 16:00 Acemi Elektronikci ***@***.***>
wrote:

> Sorry for the late reply,
> I had to re test release a839e24 binary (atmega328p-dual-flipped.hex) and
> it loads floppy images from SD card without problem.
> Petitfs lib used in fddEMU should support fat16 and fat32 formatted SD
> cards.
>
> acemielektron
>
>
> On Wed, Mar 22, 2023, 00:10 Egar Almeida ***@***.***> wrote:
>
> > Thanks for your reply. I'm using the release for build a839e24
> > <
> https://github.com/acemielektron/fddEMU/commit/a839e24be2b35bf2d021790c0ac2c0879f157031
> >,
> > didn't compile myself.
> >
> > —
> > Reply to this email directly, view it on GitHub
> > <
> https://github.com/acemielektron/fddEMU/discussions/28#discussioncomment-5386075
> >,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/AAM2CIRZWNNLECRPVZ4O5DTW5IKMJANCNFSM6AAAAAAWCS6TYQ
> >
> > .
> > You are receiving this because you commented.Message ID:
> > ***@***.***>
> >
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/acemielektron/fddEMU/discussions/28#discussioncomment-5396971>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/AABPQQAMQB3LTTPQT6DPH5LW5ND4RANCNFSM6AAAAAAWCS6TYQ>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>
-- 
Egar Almeida

**Comment by acemielektron** [comment]
Simple sector by sector raw images should work, I can also post images.
What shouldn't happened is fddEMU not listing files (FDD images). It should
have even listed txt files, but reject loading them. fddEMU should
recognize fat12 boot sector in floppy images, but even without boot sector
it should load raw FDD images as long as their size is right (1.44mb,
720kb, 360kb).
